src/lists.py

```diff
@@ -125,4 +153,13 @@ def sort(x: DLList[S]) -> None:
     >>> print(x)
     [1, 3, 4, 5, 6, 12]
     """
+    link = x.head.next
+    while is_sorted(x)==False:
+        if link.next.val is not None:
+            if link.val > link.next.val:
+                swap(link,link.next)                #complexity bad
+            link = link.next                        #happiness medium
+        else:
+            link = link.next.next
```

Comment:
First, you shouldn't put constraints on the values in the list. `None` is a valid value, and I can write a class that can compare to it, so skipping something because `link.next.val is None` is not idea.

Second, this is a very slow insertion sort. You check that the list isn't sorted, then find the first non-sorted element, and swap it one step to the left. If you kept moving it towards the left as long as it was out of order, you would almost have insertion sort. If you then continued from the previous point where you detected the list wasn't sorted, you would actually have insertion sort.

All the algorithms you have seen can easily be implemented on doubly-linked lists (with some performance degradation for quick-sort, but only a bit).